packages/app/src/server/service/search.ts

```diff
@@ -73,9 +75,19 @@ class SearchService implements SearchQueryParser, SearchResolver {
       logger.error(err);
     }
 
+    this.isElasticsearchReindexOnBoot = this.configManager.getConfig('crowi', 'app:elasticsearchReindexOnBoot');
+
     if (this.isConfigured) {
       this.fullTextSearchDelegator.init();
       this.registerUpdateEvent();
+      if (!this.isElasticsearchReindexOnBoot) {
+        logger.info('ELASTICSEARCH_REINDEX_ON_BOOT value is false, no reindex on boot');
+        this.normalizeIndices();
+      }
+      else {
+        logger.info('Reindex elasticsearch is running');
+        this.rebuildIndex(); // rebuildIndex() will call normalizeIndices() in the end of rebuildIndex() process
+      }
```

Comment:
In this case you just probably need the true condition and its process for `this.isElasticsearchReindexOnBoot`.
No need for `!this.isElasticsearchReindexOnBoot` as it's called by `this.fullTextSearchDelegator.init();`

Reply:
Done